components/dashboard/src/service/json-rpc-organization-client.ts

```diff
@@ -226,11 +227,21 @@ export class JsonRpcOrganizationClient implements PromiseClient<typeof Organizat
         if (!request.organizationId) {
             throw new ApplicationError(ErrorCodes.BAD_REQUEST, "organizationId is required");
         }
-        await getGitpodService().server.updateOrgSettings(request.organizationId, {
+        const update: Partial<OrganizationSettings> = {
             workspaceSharingDisabled: request?.workspaceSharingDisabled,
             defaultWorkspaceImage: request?.defaultWorkspaceImage,
             allowedWorkspaceClasses: request?.allowedWorkspaceClasses,
-        });
+            restrictedEditorNames: request?.restrictedEditorNames,
+        };
+        if (request.updateRestrictedEditorNames) {
+            update.restrictedEditorNames = request.restrictedEditorNames;
+        } else if (request.restrictedEditorNames && request.restrictedEditorNames.length > 0) {
+            throw new ApplicationError(
+                ErrorCodes.BAD_REQUEST,
+                "updateRestrictedEditorNames is required to be true to update restrictedEditorNames",
+            );
+        }
+        await getGitpodService().server.updateOrgSettings(request.organizationId, update);
```

Comment:
@mustard-mh forgot to add the same logic for `pinnedEditorVersions` here, could you add it pls :pray:

Reply:
Added https://github.com/gitpod-io/gitpod/pull/19542/commits/2b193cffbf7d5118c0db3f4d0ccf818e7bedb13d